src/traction_modem/Link.hxx

```diff
@@ -0,0 +1,404 @@
+/** @copyright
+ * Copyright (c) 2025, Stuart Baker
+ * All rights reserved
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file Link.hxx
+ *
+ * Implements link management and acts as a proxy to TxFlow and RxFlow.
+ *
+ * @author Stuart Baker
+ * @date 26 May 2025
+ */
+
+#ifndef _TRACTION_MODEM_LINK_HXX_
+#define _TRACTION_MODEM_LINK_HXX_
+
+#include "traction_modem/RxFlow.hxx"
+#include "traction_modem/TxFlow.hxx"
+
+namespace traction_modem
+{
+
+/// Interface for link start and up/down status.
+class LinkStatusInterface
+{
+protected:
+    /// Called when the link is started.
+    virtual void link_start()
```

Comment:
since these are event callbacks, we could consider calling these as on_link_start(), on_link_up(), on_link_down().
This eliminates confusion that this finction call would cause the link to start or go up or go down.